test/src/main/java/org/springframework/security/test/web/support/WebMvcTestUtils.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.test.web.support;
+
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.context.support.WebApplicationContextUtils;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter;
+
+/**
+ * A utility class for testing applications that use Spring Security's Web MVC support
+ *
+ * @author Josh Cummings
+ * @since 5.3
+ */
+public class WebMvcTestUtils {
+	public static void registerArgumentResolver(HttpServletRequest request, HandlerMethodArgumentResolver argumentResolver) {
```

Comment:
@rwinch I like the idea of being able to work with controllers that use `@RegisteredOAuth2AuthorizedClient`; however, I'm concerned about how well this approach will work in a multi-threaded test environment.

I could resolve it with a `synchronized` block that locks on the non-null `adapter` instance, but I wonder if you can see a better approach.

The consequence, as far as I can tell, is that multiple threads could evaluate `!containsArgumentResolver` to `true` simultaneously, and each add an instance of `argumentResolver` to the list. The tests would still work, but the resulting list of resolves seems a bit messy.